### ISSUE TITLE
Laravel v11.33.0+ in-compatible with TransformsEnums::enums()

### ISSUE THREAD

**Issue by Saifallak** [spatie/laravel-enum#108]
getting this error when using `use TransformsEnums;`

```
 Declaration of Illuminate\Http\Request::enums($key, $enumClass) must be compatible with Spatie\Enum\Laravel\Http\Requests\TransformsEnums::enums(): array
```

**Comment by Saifallak** [comment]
related to this https://github.com/laravel/framework/pull/53540

**Comment by boserup** [comment]
For those interested, as a fix (whilst transitioning away from Spatie enums in our Laravel project), we implemented a custom trait:

```php
/**
 * Fix enum method collision with Laravel v11.33.0+
 * @link https://github.com/spatie/laravel-enum/issues/108
 *
 * @method void transformEnums(array $transformations)
 */
trait TransformsLegacyEnums
{
    protected function passedValidation()
    {
        $this->transformEnums($this->legacyEnums());
    }

    abstract public function legacyEnums(): array;
}
```
